workshop/content/lab1.4_developer.md

```diff
@@ -128,5 +131,156 @@ If you wait another ~90s, then you will see the `hello-python` service again be
 kn service delete hello-python
 ```
 
+## Functions
+
+Previously, we deployed a Knative Service, which was just a containerized application.  This works just fine, but notice that we had to choose the way we exposed and handled the http endpoint.  In our example above, we chose Flask.  You can imagine for Ruby we might choose Sinatra, for Java maybe Spring Boot, and for Node maybe Express.
+
+In some use cases, it would be great if we could offload these http concerns to the platform.  That way, all we would have to do is just write a single function and just deploy that.  This is often referred to as Functions as a Service (FAAS) and this is exactly what Serverless Functions allow us to do.  For a developer, this let's us focus only on just writing the code itself and nothing else.
+
+There is one caveat to the following section, specifically the deployment section below.  You will need and be logged into an image repository like http://quay.io or http://hub.docker.com.  If you are using quay.io, go ahead and create a registry named `func` and make the visibility public by going here https://quay.io/new/.  Also log in to it by running `docker login quay.io` or `podman login quay.io`.
```

Comment:
We need to add `podman` to our Docker image.  We should probably make this run command something they could execute.

```execute
podman login quay.io`
```

(assuming they do this in homeroom)